### DISCUSSION TITLE
Auto Clean old Images / volumes and remove unused images

### DISCUSSION THREAD

**Discussion by Zoey2936** [nextcloud/all-in-one#800] [Ideas] [answered]
Every time I update Nextcloud AIO, there are still all the old images on my server, if they would automatically get removed, then this would be nice
![grafik](https://user-images.githubusercontent.com/75573284/172397235-547253cc-171c-4198-94a4-cac7afb95553.png)
(Image is from my portainer)

**Comment by Zoey2936** [comment]
![grafik](https://user-images.githubusercontent.com/75573284/172399663-1b11296c-b509-48a5-bdb4-971d74d9974a.png)
And maybe also the domaincheck / watchover container and image after they are no longer needed, when the aio stack runs

**Comment by Zoey2936** [comment]
And maybe also clean the temporary volumes of redis and onlyoffice after updating nextcloud AIO (or disable the creation of temporary volumes in their dockerfiles?)
![grafik](https://user-images.githubusercontent.com/75573284/172400216-b92ff65d-e2cf-4969-acb3-270af65d5d48.png)

**Comment by szaimen** [comment, marked as answer]
> Every time I update Nextcloud AIO, there are still all the old images on my server, if they would automatically get removed, then this would be nice

This is already implemented in a cronjob that runs once a day.

**Reply by Zoey2936** [reply]
Ok, after the next update I will wait a bit and check this, since in may opinion one time they stayes over a week and then I deleted them, thanks

**Reply by szaimen** [reply]
https://github.com/nextcloud/all-in-one/blob/b575bf40280d1ad99711d4b4beb94d96db99f075/Containers/mastercontainer/cron.sh#L95

**Reply by Zoey2936** [reply]
And whats with the temporary volumes / containers?

**Reply by szaimen** [reply]
Temporary volumes should be cleaned up upon recreation of the containers...

For containers there is indeed no logic implemented
